### DISCUSSION TITLE
Convert mapping to attributes

### DISCUSSION THREAD

**Discussion by darthf1** [doctrine/doctrine2#10211] [Support Questions] [answered]
Hi!

I'm currently on the [PHP mapping](https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/php-mapping.html), and wish to migrate to the Attribute mapping.

Mappings can be converted with the `doctrine:mapping:convert` command, but an attribute export driver is not available:
https://github.com/doctrine/orm/blob/0b14c01b93d069dcd3a0e6fe53d5fef5de83432c/lib/Doctrine/ORM/Tools/Export/ClassMetadataExporter.php#L20-L26

My question is: is there another way via Doctrine to convert the mapping? If not, since PHP Mapping will be removed in the `3.0` release, is this something planned (be)for(e) that release?

Another route would be to convert to annotations, and use rector to convert annotations to attributes.

**Comment by derrabus** [comment, marked as answer]
Use Rector.

https://www.doctrine-project.org/2022/11/04/annotations-to-attributes.html
